### PR TITLE
Fix for negative values of ebi_v in assancao_rate smoother function

### PR DESCRIPTION
**Pull request by Omarkhursheed**
Hello! Please make sure to check all these boxes before submitting a Pull Request
(PR). Once you have checked the boxes, feel free to remove all text except the
justification in point 5. 

1. [X] You have run tests on this submission, either by using [Travis Continuous Integration testing](https://github.com/pysal/pysal/wiki/GitHub-Standard-Operating-Procedures#automated-testing-w-travis-ci) testing or running `nosetests` on your changes?
2. [X] This pull request is directed to the `pysal/dev` branch.
3. [ ] This pull introduces new functionality covered by
   [docstrings](https://en.wikipedia.org/wiki/Docstring#Python) and
   [unittests](https://docs.python.org/2/library/unittest.html)? 
4. [ ] You have [assigned a
   reviewer](https://help.github.com/articles/assigning-issues-and-pull-requests-to-other-github-users/) and added relevant [labels](https://help.github.com/articles/applying-labels-to-issues-and-pull-requests/)
5. [X] The justification for this PR is: According to #909, the Assuncao rate smoother function does not work correctly for negative values of ebi_v. This PR implements a fix suggested in the issue, by setting ebi_v = ebi_b / b for negative values of ebi_v instead of ebi_a + ebi_b/ b.

### CODE REVIEW

**Comment by Omarkhursheed** [comment]
@ljwolf @darribas , I checked the build, and the checks are failing because I am supposed to use ebi_v.any() or ebi_v.all() while checking if ebi_v is less than 0, because I'd need to check an array of booleans. Which of these am I supposed to use, or do I need to change my approach to the problem?

**Comment by ljwolf** [comment]
According to the issue you refer to, it needs to be done on an element-by-element basis. I'd use a [boolean masked assignment](https://docs.scipy.org/doc/numpy/user/basics.indexing.html#boolean-or-mask-index-arrays). 

Also, I'm not read up on this particular issue, so possibly referring to the paper would clear up the suggestion in #909.

File: pysal/esda/smoothing.py [hunk 1]
```diff
@@ -550,6 +550,8 @@ def assuncao_rate(e, b):
     s2 = sum(b * ((y - ebi_b) ** 2)) / b_sum
     ebi_a = s2 - ebi_b / (float(b_sum) / len(e))
     ebi_v = ebi_a + ebi_b / b
+    if (ebi_v < 0):
```
Comment (timstallmann) [inline pysal/esda/smoothing.py]: Will this work since `ebi_v` is an array? I did
`ebi_v = np.where(ebi_v_raw < 0, ebi_b / b, ebi_v_raw)` instead here

Comment (Omarkhursheed) [inline pysal/esda/smoothing.py]: Thank you so much. This is exactly what I was looking for.

**Pull request merged**